versions/3.0.4.md

```diff
@@ -1031,7 +1031,7 @@ Field Name | Type | Description
 <a name="parameterDescription"></a>description | `string` | A brief description of the parameter. This could contain examples of use. [CommonMark syntax](https://spec.commonmark.org/) MAY be used for rich text representation.
 <a name="parameterRequired"></a>required | `boolean` | Determines whether this parameter is mandatory. If the [parameter location](#parameterIn) is `"path"`, this property is **REQUIRED** and its value MUST be `true`. Otherwise, the property MAY be included and its default value is `false`.
 <a name="parameterDeprecated"></a> deprecated | `boolean` | Specifies that a parameter is deprecated and SHOULD be transitioned out of usage. Default value is `false`.
-<a name="parameterAllowEmptyValue"></a> allowEmptyValue | `boolean` | Sets the ability to pass empty-valued parameters. This is valid only for `query` parameters and allows sending a parameter with an empty value. Default value is `false`. If [`style`](#parameterStyle) is used, and if behavior is `n/a` (cannot be serialized), the value of `allowEmptyValue` SHALL be ignored. Use of this property is NOT RECOMMENDED, as it is likely to be removed in a later revision.
+<a name="parameterAllowEmptyValue"></a> allowEmptyValue | `boolean` | If `true`, allows passing empty-valued parameters regardless of `schema` or `content`. This is valid only for `query` parameters. Default value is `false`. If [`style`](#parameterStyle) is used, and if behavior is `n/a` (cannot be serialized), the value of `allowEmptyValue` SHALL be ignored. Use of this property is NOT RECOMMENDED, as it is likely to be removed in a later revision; instead, ensure that the parameter's schema accepts any desired empty value(s).
```

Comment:
With https://editor.swagger.io/ I can only send an empty query option if it is defined with `allowEmptyValue:true`. Allowing an empty string in the schema is insufficient, the query option will not be sent if the input field is left empty.

I'd rather not recommend an alternative that doesn't work in a very prominent tool, and leave the text as it is now.

Reply:
@ralfhandl well then what are people supposed to do to allow empty values since they're not supposed to use this keyword?  We should not warp the spec because a tooling vendor got something wrong.  On the other hand, if swagger's interpretation is correct, then this never should have been discouraged because _there is no other way to send an empty value_.

So which is it?  Because we need a PR here, either to clarify that you can do the expected thing, or to allow people to use the keyword again if you _MUST_ use it to allow empty values.

Reply:
Also, are "empty values" here the undefined values of RFC6570?  Which include:  `null`, empty arrays, empty objects, and objects that only have `null` properties, but ***NOT*** the empty string?  Also, what if `content` is used instead of `style`?  This keyword is outside of the `style`-specific block so I guess that means if you end up with a zero-length value?  What if your value in `content` is a value that RFC6570, which does not appear to apply to `content`, considers undefined?  Is that how you get around RFC6570's undefined definitions and serialize an empty object?

Reply:
Thinking about this some more I believe that "allow empty values" is a UI thingy, needed to allow sending a query option with an empty value instead of omitting the "unfilled" query option:

![image](https://github.com/OAI/OpenAPI-Specification/assets/951576/a24321f8-20d8-411b-be4d-52ba23f494ae)

UIs REALLY SHOULD render such a check box if (and only if) the parameter's schema accepts the empty string, or provide some other means for users to express sending the query option with nothing after the `=` character.